modules/notifications/notifications.php

```diff
@@ -420,6 +420,7 @@ function ajax_save_post_subscriptions() {
 		}
 
 		if ( 'ef-selected-users[]' === $_POST['ef_notifications_name'] ) {
+			add_filter( 'ef_notification_auto_subscribe_current_user', '__return_false', PHP_INT_MAX );
```

Comment:
It would be useful to add a comment here noting why we're doing this.

```suggestion
			// Prevent auto-subscribing users that have opted out of notifications.
			add_filter( 'ef_notification_auto_subscribe_current_user', '__return_false', PHP_INT_MAX );
```